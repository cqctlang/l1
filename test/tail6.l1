// same as tail3, but use 64-bit result.
fac = @lambda(n,rv){
	if(n <= 1)
		return rv;
	else{
		printf("%u\n", rv);
		return fac(n-1,n*rv);
	}
};
print(fac(50,1ULL));
