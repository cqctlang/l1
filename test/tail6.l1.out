entry: var params, 1 locals, 0 temps
L4: 2 params, 0 locals, 0 temps
1
50
2450
117600
5527200
254251200
11441304000
503417376000
21646947168000
909171781056000
37276043023296000
1491041720931840000
2810394895213105152
14561285649550237696
3811990895781797888
8104463732177862656
6955069520581918720
15111434815270617088
615258913772470272
1241541167009497088
1594288029875306496
10935152748840091648
3524780463300280320
6460132603860090880
8402883640836489216
15560789850943651840
1638120725690712064
2421409269157986304
352180969505030144
7747981329110663168
15133655321647513600
7525201253597446144
13851615302384615424
9521402484698906624
14289889650204999680
7277305518765375488
16925862412932874240
15601144896545619968
18347442917997543424
17255130205165453312
5338991519724470272
16496427049825599488
893890858753982464
7151126870031859712
13164399942803914752
5199423361985282048
7550372736216858624
11754746871157882880
16817496539764097024
<cval 15188249005818642432>
